refinery_core/src/util.rs

```diff
@@ -58,9 +59,34 @@ pub fn find_migration_files(
     Ok(file_paths)
 }
 
+/// Loads SQL migrations from a path. This enables dynamic migration discovery, as opposed to
+/// embedding. The resulting collection is ordered by version.
+pub fn load_sql_migrations(location: impl AsRef<Path>) -> Result<Vec<Migration>, Error> {
+    let migration_files = find_migration_files(location, MigrationType::Sql)?;
+
+    let mut migrations = vec![];
+
+    for path in migration_files {
+        let sql = std::fs::read_to_string(path.as_path())
+            .map_err(|e| Error::new(Kind::InvalidMigrationFile(path.to_owned(), e), None))?;
```

Comment:
Sorry, I was probably not clear, [`read_to_string`](https://doc.rust-lang.org/std/fs/fn.read_to_string.html) may return 2 errors:
>This function will return an error if path does not already exist. Other errors may also be returned according to [OpenOptions::open](https://doc.rust-lang.org/std/fs/struct.OpenOptions.html#method.open).
>
> If the contents of the file are not valid UTF-8, then an error will also be returned.

Since we already have `InvalidMigrationPath`, I'd distinguish between it and `InvalidMigrationFile`, i.e. when `read_to_string` returns [`NotFound`](https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.NotFound) return `InvalidMigrationPath` else return `InvalidMigrationFile` wdyt?

Reply:
Sounds good, changed.